dbt_semantic_interfaces/parsing/text_input/rendering_helper.py

```diff
@@ -0,0 +1,224 @@
+        item_type = QueryItemType.METRIC
+        allowed_methods = self._valid_method_mapping[item_type]
+
+        def _create(metric_name: str, group_by: Sequence[str] = ()) -> _RenderingClassForJinjaTemplate:
+            return _RenderingClassForJinjaTemplate(
+                description_processor=description_processor,
+                allowed_methods=allowed_methods,
+                initial_item_description=QueryItemDescription(
+                    item_type=item_type,
+                    item_name=metric_name,
+                    entity_path=(),
+                    time_granularity_name=None,
+                    date_part_name=None,
+                    group_by_for_metric_item=tuple(group_by),
+                    descending=None,
+                ),
+            )
+
+        return _create
+
+
+class _RenderingClassForJinjaTemplate:
+    """Helper class that behaves like a builder object as used in a Jinja template.
+
+    e.g. in the Jinja template:
+
+        {{ Dimension('listing__created_at').grain('day').date_part('month') }}
+
+    The `Dimension('listing__created_at')` is an instance of this class and when builder methods like `.grain()` are
+    called on it, the state of the instance is updated and returns itself so that additional builder methods can be
+    chained.
+    """
+
+    def __init__(
+        self,
+        description_processor: QueryItemDescriptionProcessor,
+        allowed_methods: FrozenSet[ObjectBuilderMethod],
+        initial_item_description: QueryItemDescription,
+    ) -> None:
+        """Initializer.
+
+        Args:
+            description_processor: The description processor that will run using the query-item description described
+            in the builder call. It will run after all builder methods are called.
+            allowed_methods: Builder methods that can be used. Otherwise, an `InvalidBuilderMethodException` is raised.
+            initial_item_description: The starting description. Usually it contains the element name and entity path.
+        """
+        self._description_processor = description_processor
+        self._allowed_builder_methods = allowed_methods
+        self._current_description = initial_item_description
+
+    def _update_current_description(
+        self,
+        time_granularity_name: Optional[str] = None,
+        date_part_name: Optional[str] = None,
+        descending: Optional[bool] = None,
+    ) -> None:
+        args = (time_granularity_name, date_part_name, descending)
+        assert sum(1 for arg in args if arg is not None) == 1, f"Expected exactly 1 argument set, but got {args}"
+
+        builder_method: Optional[ObjectBuilderMethod]
+        if time_granularity_name is not None:
+            builder_method = ObjectBuilderMethod.GRAIN
+        elif date_part_name is not None:
+            builder_method = ObjectBuilderMethod.DATE_PART
+        elif descending is not None:
+            builder_method = ObjectBuilderMethod.DESCENDING
```

Comment:
Can we somehow remove the necessity for this block? I think we should strive to reduce the amount of boilerplate necessary when adding a builder method.

Reply:
Yeah, this block can be removed. However, I don't think net boiler plate will change - did you have a particular approach in mind?

Reply:
Yeah, this block can be removed. However, I don't think net boiler plate will change - did you have a particular approach in mind?

Reply:
Added a commit with that update.

Reply:
The change you made is great thank you! Now if we add another builder method, we don't have to add another `elif`.